pkg/apis/core/types.go

```diff
@@ -5459,6 +5459,18 @@ type Preconditions struct {
 	UID *types.UID
 }
 
+// LogStreamType represents the desired log stream type.
+type LogStreamType string
+
+const (
+	// LogStreamTypeStdout is the stream type for stdout.
+	LogStreamTypeStdout LogStreamType = "stdout"
+	// LogStreamTypeStderr is the stream type for stderr.
+	LogStreamTypeStderr LogStreamType = "stderr"
+	// LogStreamTypeAll represents the combined stdout and stderr.
+	LogStreamTypeAll LogStreamType = "all"
```

Comment:
I'd reconsider why we want an explicit "all"?  For backwards compat, if you do not pass this `stream=` param, you will get "all".  The way I see it, if you DO pass this new param, you are explicitly requesting a single stream.

Reply:
I like having explicit option alongside the default empty as it allows to keep the key around with the explicit value, so it is easy to change it to other explicit value. If you need to change value to empty or remove the kay completely, it makes the experience of trying different options harder.

Reply:
My two cents: let's consider the implementation of kubectl, if we had an explicit `LogStreamTypeAll`, we could easily map the user choice to these constants. For instance:
```go
var stream LogStreamType
switch flags.Stream {
// ...
}
clientset.Pods(ns).GetLogs(container, PodLogOptions{
  Stream: &stream,
}
```

However, if we didn't have an explicit `LogStreamTypeAll`, i.e. the `stream` had to be omitted, it would be harder to set the field. For instance:
```
logOpts := PodLogOptions{}
switch flags.Stream {
case "stdout":
  logOpts.Stream = ptr.To(LogStreamTypeStdout)
// ...
case "all":
  logOpts.Stream = nil
}
```

Reply:
I don't mind keeping it, I guess, it just seems superfluous to me, since this is not stored anywhere.